domain/user/service/service.go

```diff
@@ -233,47 +321,54 @@ func (s *Service) RemoveUser(ctx context.Context, name string) error {
 // - internal/auth.ErrPasswordNotValid: If the password supplied is not valid.
 func (s *Service) SetPassword(
 	ctx context.Context,
-	name string,
+	uuid user.UUID,
 	password auth.Password,
 ) error {
 	defer password.Destroy()
-	if err := ValidateUsername(name); err != nil {
-		return fmt.Errorf("username %q: %w", name, err)
+	if err := uuid.Validate(); err != nil {
+		return fmt.Errorf("setting password for user uuid %q: %w", uuid, err)
+	}
+
+	usr, err := s.GetUser(ctx, uuid)
+	if err != nil {
+		return fmt.Errorf("setting password for user uuid %q: %w", uuid, err)
 	}
 
 	salt, err := auth.NewSalt()
 	if err != nil {
-		return fmt.Errorf("setting password for user %q, generating password salt: %w", name, err)
+		return fmt.Errorf("setting password for user %q with uuid %q, generating password salt: %w", usr.Name, usr.UUID.String(), err)
 	}
 
 	pwHash, err := auth.HashPassword(password, salt)
 	if err != nil {
-		return fmt.Errorf("setting password for user %q, hashing password: %w", name, err)
+		return fmt.Errorf("setting password for user %q with uuid %q, hashing password: %w", usr.Name, usr.UUID.String(), err)
 	}
 
-	if err = s.st.SetPasswordHash(ctx, name, pwHash, salt); err != nil {
-		return fmt.Errorf("setting password for user %q: %w", name, err)
+	if err = s.st.SetPasswordHash(ctx, usr.UUID, pwHash, salt); err != nil {
+		return fmt.Errorf("setting password for user %q withb uuid %q: %w", usr.Name, usr.UUID.String(), err)
 	}
 	return nil
 }
 
 // ResetPassword will remove any active passwords for a user and generate a new
 // activation key for the user to use to set a new password.
-// The following error types are possible from this function:
-// - usererrors.UsernameNotValid: When the username supplied is not valid.
-// - usererrors.NotFound: If no user by the given name exists.
-func (s *Service) ResetPassword(ctx context.Context, name string) ([]byte, error) {
-	if err := ValidateUsername(name); err != nil {
-		return nil, fmt.Errorf("username %q: %w", name, err)
+func (s *Service) ResetPassword(ctx context.Context, uuid user.UUID) ([]byte, error) {
+	if err := uuid.Validate(); err != nil {
+		return nil, fmt.Errorf("resetting password for user uuid %q: %w", uuid, err)
+	}
+
+	usr, err := s.GetUser(ctx, uuid)
```

Comment:
Same comments as above. I don't think we need to hit the database twice. In state SetActivationKey we can infer if the user doesn't exist and return the appropriate error.

Reply:
thx, removed

PS: this is an artifact from the "Name-centric" implementation